test/parallel/test-debug-port-cluster.js

```diff
@@ -16,12 +16,12 @@ child.stderr.setEncoding('utf8');
 
 let stderr = '';
 child.stderr.on('data', (data) => {
+  process.stderr.write(`[DATA] ${data}`);
   stderr += data;
-  if (child.killed !== true && stderr.includes('all workers are running'))
+  if (child.killed !== true && stderr.includes('all workers are running')) {
     child.kill();
-});
-
-process.on('exit', () => {
-  for (let port = PORT_MIN; port <= PORT_MAX; port += 1)
-    assert(stderr.includes(`Debugger listening on port ${port}`));
+    for (let port = PORT_MIN; port <= PORT_MAX; port += 1) {
+      assert(stderr.includes(`Debugger listening on port ${port}`));
+    }
+  }
```

Comment:
@bnoordhuis Ah, yes, good point. I've extracted the `for` loop to its own function and wrapped it in `common.mustCall()` so there will be an at-exit check that the function ran. Stress testing now to see if that re-introduces the problem or not. (If it does, I guess my "solution" was to not check when the problem arose. :-(  )